applications/GeoMechanicsApplication/custom_constitutive/truss_backbone_constitutive_law.cpp

```diff
@@ -0,0 +1,166 @@
+        rValue = mAccumulatedStrain;
+    } else {
+        KRATOS_ERROR << "Can't get the specified value for " << rThisVariable << std::endl;
+    }
+    return rValue;
+}
+
+array_1d<double, 3>& TrussBackboneConstitutiveLaw::GetValue(const Variable<array_1d<double, 3>>& rThisVariable,
+                                                            array_1d<double, 3>& rValue)
+{
+    KRATOS_ERROR << "Can't get the specified value for " << rThisVariable << std::endl;
+}
+
+void TrussBackboneConstitutiveLaw::SetValue(const Variable<double>& rThisVariable,
+                                            const double&           rValue,
+                                            const ProcessInfo&      rCurrentProcessInfo)
+{
+    if (rThisVariable == ACCUMULATED_STRAIN) {
+        mAccumulatedStrain = rValue;
+    } else {
+        KRATOS_ERROR << "Can't set the specified value for " << rThisVariable << std::endl;
+    }
+}
+
+double& TrussBackboneConstitutiveLaw::CalculateValue(ConstitutiveLaw::Parameters& rParameterValues,
+                                                     const Variable<double>&      rThisVariable,
+                                                     double&                      rValue)
+{
+    if (rThisVariable == TANGENT_MODULUS) {
+        const auto youngs_modulus = rParameterValues.GetMaterialProperties()[YOUNG_MODULUS];
+        const auto axial_strain(rParameterValues.GetStrainVector()[0]);
+        if (IsWithinUnReLoading(axial_strain, youngs_modulus)) {
+            rValue = youngs_modulus;
+        } else {
+            rValue = BackboneStiffness(axial_strain);
+        }
+    } else {
+        KRATOS_ERROR << "Can't calculate the specified value" << std::endl;
+    }
+    return rValue;
+}
+
+void TrussBackboneConstitutiveLaw::CalculateMaterialResponsePK2(Parameters& rValues)
+{
+    // backbone loading, un- and reloading laws
+    const auto axial_strain(rValues.GetStrainVector()[0]);
+    auto&      axial_stress_vector = rValues.GetStressVector();
+    const auto youngs_modulus      = rValues.GetMaterialProperties()[YOUNG_MODULUS];
+    if (IsWithinUnReLoading(axial_strain, youngs_modulus)) {
+        // Un- or reloading
+        axial_stress_vector[0] = youngs_modulus * (axial_strain - mUnReLoadCenter);
+    } else {
+        // backbone
+        axial_stress_vector[0] =
+            BackboneStress(mAccumulatedStrain + (std::abs(axial_strain - mUnReLoadCenter) -
+                                                 (CalculateUnReLoadAmplitude(youngs_modulus) / 2.)));
+        axial_stress_vector[0] = (axial_strain - mUnReLoadCenter) >= 0. ? axial_stress_vector[0]
+                                                                        : -axial_stress_vector[0];
+    }
+    rValues.SetStressVector(axial_stress_vector);
+}
+
+void TrussBackboneConstitutiveLaw::FinalizeMaterialResponsePK2(Parameters& rValues)
+{
+    CalculateMaterialResponsePK2(rValues);
+    const auto axial_strain(rValues.GetStrainVector()[0]);
```

Comment:
In line with my earlier comments:
```suggestion
    const auto axial_strain = rValues.GetStrainVector()[0];
```

Reply:
Done.